__tests__/app.test.js

```diff
@@ -145,4 +145,83 @@ describe('6', () => {
                 }))
         })
     });
+});
+
+describe('9. GET /api/articles/:article_id/comments', () => {
+    test('should return 200 and comments', () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({body}) => {
+            expect(body.comments).toBeInstanceOf(Array)
+            expect(body.comments).toHaveLength(11)
+            body.comments.forEach(comment => {
+                expect(comment).toEqual(
+                    expect.objectContaining({
+                        comment_id: expect.any(Number),
+                        body: expect.any(String),
+                        article_id: 1,
+                        author: expect.any(String),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String)
+                    })
+                )
+            })
+        })
+    });
+    test('should return 400 and bad request for invalid article id', () => {
+        return request(app)
+        .get("/api/articles/banana/comments")
+        .expect(400)
+        .then(({body}) => {
+            expect(body.message).toBe("bad request")
+        })
+    });
+    test('should return 404 if article has no comments', () => {
+        return request(app)
+        .get("/api/articles/10/comments")
+        .expect(404)
+        .then(({body}) => {
+            expect(body.message).toBe("no comments found")
+        })
+    });
+});
+
+describe('10. POST /api/articles/:article_id/comments', () => {
+    test('return 201 and new comment', () => {
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send({username: "icellusedkars", body:"string of text"})
+        .expect(201)
+        .then(({body}) => {
+            expect(body.comment).toEqual(
+                expect.objectContaining({
+                    comment_id: expect.any(Number),
+                    body: expect.any(String),
+                    article_id: 1,
+                    author: expect.any(String),
+                    votes: 0,
+                    created_at: expect.any(String)
+                })
+            )
+        })
+    });
+    test('return 400 and bad request with invalid input', () => {
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send({notusername: "icellusedkars", body:"string of text"})
+        .expect(400)
+        .then(({body}) => {
+            expect(body.message).toBe("invalid new comment")
+        })
+    });
+    test('return 404 and bad request with invalid article id', () => {
```

Comment:
If you are checking for an error on the id you should send a valid request body so you can be sure that the error is due to the id. Also the description isn't quite right as the article id is valid, it just doesn't exist